beacon-chain/cache/attestation_data.go

```diff
@@ -3,169 +3,48 @@ package cache
 import (
 	"context"
 	"errors"
-	"fmt"
-	"math"
 	"sync"
-	"time"
 
-	"github.com/prometheus/client_golang/prometheus"
-	"github.com/prometheus/client_golang/prometheus/promauto"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
 	ethpb "github.com/prysmaticlabs/prysm/v4/proto/prysm/v1alpha1"
-	"k8s.io/client-go/tools/cache"
 )
 
-var (
-	// Delay parameters
-	minDelay    = float64(10)        // 10 nanoseconds
-	maxDelay    = float64(100000000) // 0.1 second
-	delayFactor = 1.1
-
-	// Metrics
-	attestationCacheMiss = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "attestation_cache_miss",
-		Help: "The number of attestation data requests that aren't present in the cache.",
-	})
-	attestationCacheHit = promauto.NewCounter(prometheus.CounterOpts{
-		Name: "attestation_cache_hit",
-		Help: "The number of attestation data requests that are present in the cache.",
-	})
-	attestationCacheSize = promauto.NewGauge(prometheus.GaugeOpts{
-		Name: "attestation_cache_size",
-		Help: "The number of attestation data in the attestations cache",
-	})
-)
-
-// ErrAlreadyInProgress appears when attempting to mark a cache as in progress while it is
-// already in progress. The client should handle this error and wait for the in progress
-// data to resolve via Get.
-var ErrAlreadyInProgress = errors.New("already in progress")
+type AttestationConsensusData struct {
+	Slot             primitives.Slot
+	HeadRoot         []byte
+	TargetCheckpoint *ethpb.Checkpoint
+	SourceCheckpoint *ethpb.Checkpoint
```

Comment:
Why are these pointers?

Reply:
I don't want to copy the whole protobuf object; there are many irrelevant fields. 
Instead of Checkpoint, I'll switch to the root and epoch

Reply:
You can use `forkchoicetypes.Checkpoint` for this